src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -818,12 +818,27 @@ internal bool HasCollectionExpressionApplicableAddMethod(SyntaxNode syntax, Type
             {
                 return element switch
                 {
-                    BoundCollectionElementInitializer collectionInitializer => collectionInitializer.Arguments[collectionInitializer.InvokedAsExtensionMethod ? 1 : 0],
+                    BoundCollectionElementInitializer collectionInitializer => getCollectionInitializerElement(collectionInitializer),
                     BoundDynamicCollectionElementInitializer dynamicInitializer => dynamicInitializer.Arguments[0],
                     _ => null,
                 };
             }
             return element;
+
+            static BoundExpression getCollectionInitializerElement(BoundCollectionElementInitializer collectionInitializer)
+            {
+                int argIndex = collectionInitializer.InvokedAsExtensionMethod ? 1 : 0;
+                var arg = collectionInitializer.Arguments[argIndex];
+                if (collectionInitializer.Expanded && collectionInitializer.AddMethod.Parameters[argIndex].IsParams)
```

Comment:
>&& collectionInitializer.AddMethod.Parameters[argIndex].IsParams [](http://example.com/codeflow?start=51&length=64)

I think we should remove this check. At best it is redundant. because overload resolution already checked for that. At worse, it will lead to the same bug when the ```collectionInitializer.AddMethod``` is an override that doesn't have `ParamArray` attribute (overload resolution checks for it on the least derived declaration). The ```collectionInitializer.Expanded``` condition is sufficient in my opinion.

Reply:
It's necessary to check that the `argIndex` refers to the `params` parameter even if the `collectionInitializer.Expanded` to allow for the case where there is a parameter before the `params` parameter (see `Add_ParamsArray_04`):
```
public void Add(T x, params T[] y) { ... }
```

Reply:
It looks like the `ParamArray` attribute is added to overrides, at least for overrides declared in C#. But I've changed the check here to avoid using `IsParams`.

Reply:
> It looks like the `ParamArray` attribute is added to overrides, at least for overrides declared in C#. 

What is important is the fact that the presence of the attribute on overrides is not required for the method to be used in expanded form. And metadata like that can easily be created.